projects/snapshot_bot/snapshot_bot.cc

```diff
@@ -0,0 +1,529 @@
+                    m_LogFile << ((Seconds)(currentTime - m_RecordingStartTime)).count() << ", ";
+
+                    // Write memory-specific CSV logging
+                    m_Memory->writeCSVLine(m_LogFile);
+
+                    // If vicon tracking is available
+                    if(m_Config.shouldUseViconTracking()) {
+                        // Get tracking data
+                        auto objectData = m_ViconTracking.getObjectData(0);
+                        const auto &position = objectData.getPosition<units::length::millimeter_t>();
+                        const auto &attitude = objectData.getAttitude<units::angle::degree_t>();
+
+                        // Write extra logging data
+                        m_LogFile << ", " << objectData.getFrameNumber() << ", " << position[0] << ", " << position[1] << ", " << position[2] << ", " << attitude[0] << ", " << attitude[1] << ", " << attitude[2];
+                    }
+
+                    // If we should save diagnostics when testing
+                    if(m_Config.shouldSaveTestingDiagnostic()) {
+                        const std::string filename = "test" + m_Config.getTestingSuffix() + "_" + std::to_string(m_TestImageIndex++) + ".png";
+                        m_LogFile << ", " << filename;
+                        // Build path to test image and save
+                        const auto testImagePath = m_Config.getOutputPath() / filename;
+                        cv::imwrite(testImagePath.str(), m_Unwrapped);
+                    }
+
+                    m_LogFile << std::endl;
+
+                    // If we should stream output
+                    /*if(m_Config.shouldStreamOutput()) {
+                        // Attempt to dynamic cast memory to a perfect memory
+                        PerfectMemory *perfectMemory = dynamic_cast<PerfectMemory*>(m_Memory.get());
+                        if(perfectMemory != nullptr) {
+                            // Get matched snapshot
+                            const cv::Mat &matchedSnapshot = perfectMemory->getBestSnapshot();
+
+                            // Calculate difference image
+                            cv::absdiff(matchedSnapshot, m_Unwrapped, m_DifferenceImage);
+
+                            char status[255];
+                            sprintf(status, "Angle:%f deg, Min difference:%f", degree_t(perfectMemory->getBestHeading()).value(), perfectMemory->getLowestDifference());
+                            cv::putText(m_DifferenceImage, status, cv::Point(0, m_Config.getUnwrapRes().height -20),
+                                        cv::FONT_HERSHEY_COMPLEX_SMALL, 1.0, 0xFF);
+
+                            // Send annotated difference image
+                            //m_NetSink.sendFrame(m_DifferenceImage);
+                        }
+                        else {
+                            LOGW << "WARNING: Can only stream output from a perfect memory";
+                        }
+                    }*/
```

Comment:
I'm guessing this is commented out because the you're getting compiler errors for the networking code?

Reply:
Yeah - I feel the solution lies on another branch as you had it working with the stone CX

Reply:
I *think* the only issue is that at some point in the distant past I changed the ``Server`` class so that it can accept multiple connections. You just need to call ``waitForConnection()`` and you get a ``Connection`` object out.

Reply:
It also no longer seems to have a ``runInBackground()`` method

Reply:
But the ``Connection`` object you get out from it will do. The ``Server`` object just listens for incoming connections now.

You could even probably just initialise a connection object like so in your ctor: ``m_Connection(Net::Server{}.waitForConnection())``

Then you just use the ``Connection`` as you're using your ``Server`` now.